### PR TITLE
to_html show bin edge label on dims

### PR DESCRIPTION
**Pull request by dtasev**
Shows the bin-edge label if the coordinate dimension is bin-edges. Should work on ND bin-edges. Here's a sample for 3D bin-edge:

```
d = sc.Dataset(
    {
    'alice': sc.Variable([Dim.Z, Dim.Y, Dim.X], values=np.random.rand(10, 10, 10),
                         variances=0.1*np.random.rand(10, 10, 10)),
    'bob': sc.Variable([Dim.X, Dim.Z], values=np.arange(0.0, 10.0, 0.1).reshape(10, 10),
                       variances=0.1*np.random.rand(10, 10))
    },
    coords={
        Dim.Z: sc.Variable([Dim.Z, Dim.Y, Dim.X], values=np.arange(0, 11*11*11).reshape(11, 11,11), unit=sc.units.m),
    })

d
```

### CODE REVIEW

File: python/src/scipp/table_html/formatting_html.py [hunk 1]
```diff
@@ -171,37 +171,69 @@ def _icon(icon_name):
...
+
+    :param is_bin_edges: If true it will add the [bin-edges] variable
     """
     cssclass_idx = " class='xr-has-index'" if is_index else ""
     dims_text = ', '.join(
-        escape(f'{str(dim)} [sparse]' if dim == var.sparse_dim else str(dim))
+        f'{str(dim)}{_make_dim_labels(dim, var.sparse_dim, bin_edges)}'
         for dim in var.dims)
-    dims_str = f"({dims_text})"
+    dims_str = escape(f"{dims_text}".replace("Dim.", ""))
```
Comment (SimonHeybrock) [inline python/src/scipp/table_html/formatting_html.py]: Saves space, yes, but doesn't it make things more confusing?

Comment (dtasev) [reply python/src/scipp/table_html/formatting_html.py]: What about showing () for no dims, and no parenthesis for other dims?
image

Comment (SimonHeybrock) [reply python/src/scipp/table_html/formatting_html.py]: I think this makes it less obvious for an uninformed user. By having parenthesis *everywhere* it is semi-obvious that *empty parentheses* mean an empty list/tuple.

**Comment by dtasev** [comment]
Seems to have an issue with the nested attributes, do not merge yet

**Comment by dtasev** [comment]
Should be good to go. To test the case that the latest commit fixes:

```
import scipp as sc
from scipp import Dim
import numpy as np

sparse = sc.Variable([Dim.Y, Dim.Z], shape=(3, sc.Dimensions.Sparse))

sparse.values[0].extend(np.arange(0))
sparse.values[2].extend(np.arange(0))
sparse.values[1].extend(np.arange(0))

d = sc.Dataset()
d['a'] = sc.Variable([Dim.Y, Dim.X, Dim.Z], shape=(3, 2, 4), variances=True)
d['b'] = sc.DataArray(sparse, coords={
    Dim.Y: sc.Variable([Dim.Y], values=np.arange(4)),
    Dim.Z: sc.Variable([Dim.Z], shape=(sc.Dimensions.Sparse,)),
})

d
```

File: python/src/scipp/table_html/formatting_html.py [hunk 2]
```diff
@@ -173,15 +173,41 @@ def _icon(icon_name):
...
+    vars_li = "".join(
+        "<li class='xr-var-item'>"
+        f"{summarize_coord(dim, var, ds)}"
+        "</span></li>"
+        for dim, var in coords)
+
+    return f"<ul class='xr-var-list'>{vars_li}</ul>"
+
+
+def summarize_labels_masks(data):
```
Comment (SimonHeybrock) [inline python/src/scipp/table_html/formatting_html.py]: Why the special treatment for coords? There are in principle no restrictions about what can be on the bin edges, i.e, you should handle labels and masks in the same way (and also data).

Comment (dtasev) [reply python/src/scipp/table_html/formatting_html.py]: 👍 I removed that func, now they all go through `summarize_coords`

**Pull request merged**